### PR TITLE
Adding instructions for new contributors

### LINKED ISSUE #158: Add reference to docs/requirements.txt
In `docs/user/contributing.rst` we should point people to the doc specific requirements

### PR DESCRIPTION
**Pull request by calebrob6**
We should make it easy as possible for people to contribute. This PR adds a new section in the README to help new first-time contributors.

@ashnair1 am I missing anything in the pre-commit section? also, thanks again for the SpaceNet and pre-commit PRs!

@adamjstewart @isaaccorley, feel free to edit/suggest as you see fit. The name of the game is to reduce the friction between _anyone_ saying, "hey I could add < x > to torchgeo," and us merging a well formatted pull request.

Closes #158

### CODE REVIEW

**Review by adamjstewart — CHANGES_REQUESTED** [review]
We should instead add this information to our documentation, I want the README to be as lightweight as possible. See https://spack.readthedocs.io/en/latest/contribution_guide.html for another example.

**Comment by ashnair1** [comment]
@calebrob6 Happy to help!

One note I would add is regarding running `pre-commit run --all-files`.

Once pre-commit hooks are installed, it only runs on changed files so it's recommended to run the hooks against all of the files when adding new hooks via `pre-commit run --all-files`. Ref: [here](https://pre-commit.com/#4-optional-run-against-all-the-files)

+1 for adding this information to the documentation or a `CONTRIBUTING.md`

**Comment by calebrob6** [comment]
@adamjstewart, we can put this wherever, no big deal. I just want to agree on the text, e.g. should we add a section for "how to set up a development environment"?

**Comment by adamjstewart** [comment]
We definitely want a section on setting up TorchGeo for development. I was planning on splitting the "Installation" docs into "installing a stable release" and "installing a development version", so that will cover most of the "how to install developer tools" stuff. Other than that, the contribution docs should mention things like:

* Your contributions must have unit tests that cover them. PRs can't be merged without 100% test coverage
* Dataset tests can't contain real images, pointer to docs on how to create fake images
* TorchGeo does not accept datasets/transforms/models that aren't specific to geospatial data, generic computer vision datasets/transforms/models should go in torchvision/Kornia/etc.

I'm sure there's more things I'm missing here.

**Comment by calebrob6** [comment]
So we need to wait until the first release for splitting the "Installation" docs into "installing a stable release" and "installing a development version". Do you want to edit the text in the README to reflect the points above, then when we have the first release we can move this to the docs and add pointers?

**Comment by adamjstewart** [comment]
I would rather add it directly to the docs than to the README. We can still have an Installation doc page that describes how to install development versions. I was waiting until we finish the paper to document all these things since that deadline is rapidly approaching. I vote we wait on this.

File: docs/user/contributing.rst
Comment (isaaccorley) [inline docs/user/contributing.rst:124]: Seems to be an extra underscore at the end? Not sure if that breaks anything.

**Review by isaaccorley — DISMISSED** [review]
lgtm

File: docs/user/contributing.rst
Comment (isaaccorley) [inline docs/user/contributing.rst:124]:
```suggestion
Now, every time you run ``git commit``, pre-commit will run and let you know if any of the files that you changed fail the linters. If pre-commit passes then your code should be ready (style-wise) for a pull request. Note that you will need to run ``pre-commit run --all-files`` if any of the hooks in ``.pre-commit-config.yaml`` change, see `here <https://pre-commit.com/#4-optional-run-against-all-the-files>`_.
```

Comment (adamjstewart) [reply docs/user/contributing.rst:124]: It's actually required: https://stackoverflow.com/questions/27420317/restructured-text-rst-http-links-underscore-vs-use

reStructuredText is one of the worst markup languages I've ever seen IMO.

Comment (adamjstewart) [reply docs/user/contributing.rst:124]: See above, this is required.

**Review by adamjstewart — DISMISSED** [review]
Approving to override my previous change request. Will wait for others to approve before merging.

File: docs/user/contributing.rst
Comment (calebrob6) [inline docs/user/contributing.rst:129]: (these can also be found in `docs/requirements.txt`)

**Review by calebrob6 — COMMENTED** [review]
I can't approve my own pull request, however think this pull request is a wonderful idea and should be incorporated with haste :)

File: docs/user/contributing.rst
Comment (calebrob6) [reply docs/user/contributing.rst:129]: @adamjstewart, I made an issue for me to add a pointer to `docs/requirements.txt` here once we merge everything to main

Comment (adamjstewart) [reply docs/user/contributing.rst]: Done

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**
